velox/exec/HashProbe.h

```diff
@@ -327,6 +331,15 @@ class HashProbe : public Operator {
   // of an invalidated table as we always spill the entire table.
   std::optional<RowColumn::Stats> columnStats(int32_t columnIndex) const;
 
+  // Returns the raw pointer to the output row mapping buffer.
+  // Throws exception if the buffer is not mutable.
+  // Note: Use this method close to the block that modifies the buffer to ensure
+  // mutability does not change in-between.
+  vector_size_t* getRawMutableOuputRowMapping() {
+    VELOX_CHECK(outputRowMapping_->isMutable());
```

Comment:
This is check is failing because we keep 2 references: One in `outputRowMapping_` and one in filter input.  Maybe loose the condition a little bit?